File: python/crear_factura.py
import json
import time
from datetime import date

from config import api_request


def main() -> None:
    payload = {
        "tipo_comprobante": "factura_b",
        "concepto": "productos",
        "fecha": date.today().isoformat(),
        "referencia_externa": f"venta_python_{int(time.time())}",
        "cliente": {
            "documento_tipo": "dni",
            "documento_numero": "30111222",
            "razon_social": "Cliente Demo",
            "condicion_iva_receptor_id": 5,
        },
        "items": [
            {
                "descripcion": "Producto de ejemplo",
                "cantidad": 1,
                "precio_unitario": 10000,
                "tipo_impuesto": "gravado",
                "iva": 21,
            }
        ],
        "total": 12100,
        "moneda": "PES",
    }

    response = api_request(
        "POST",
        "/api/comprobantes",
        headers={
            "Content-Type": "application/json",
        },
        json=payload,
    )

    print(json.dumps(response.json(), indent=2, ensure_ascii=False))


if __name__ == "__main__":
    main()

File: php/crear_factura.php
<?php

declare(strict_types=1);

require __DIR__ . '/config.php';

$payload = [
    'tipo_comprobante' => 'factura_b',
    'concepto' => 'productos',
    'fecha' => date('Y-m-d'),
    'referencia_externa' => 'venta_php_' . time(),
    'cliente' => [
        'documento_tipo' => 'dni',
        'documento_numero' => '30111222',
        'razon_social' => 'Cliente Demo',
        'condicion_iva_receptor_id' => 5,
    ],
    'items' => [
        [
            'descripcion' => 'Producto de ejemplo',
            'cantidad' => 1,
            'precio_unitario' => 10000,
            'tipo_impuesto' => 'gravado',
            'iva' => 21,
        ],
    ],
    'total' => 12100,
    'moneda' => 'PES',
];

$response = apiRequest('POST', '/api/comprobantes', $payload);
printJson($response['body']);
